### PR TITLE
added User CRUD

### PR DESCRIPTION
**Pull request by TouqeerAli**

### CODE REVIEW

**Comment by govindkmk** [comment]
Good!
You have to create enum of status in constants.

File: src/daoimp/UserDAOImpl.java
Comment (Ahmadqazi584) [inline src/daoimp/UserDAOImpl.java:1]: Brother I think Query should be placed in Dao package interfaces respectively.

**Review by Ahmadqazi584 — COMMENTED** [review]
check DaoImpl Change for Queries

**Review by govindkmk — APPROVED** [review]

**Review by MuhammadHaseebOdho — APPROVED** [review]

File: src/daoimp/UserDAOImpl.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    private final String UPDATE_QUERY = "UPDATE user SET first_Name = ?,last_Name = ?,phone_number = ?,email = ?,user_password=?,modified_date=? where id = ?";
+    private final String UPDATE_STATUS = "update user set bool_active = false where id = ?";
+
+    @Override
+    public boolean addUser(User user) {
+        boolean success = false;
+
+        try {
+            Connection con = DbConnection.getConnection();
+            PreparedStatement ps = con.prepareStatement(INSERT_QUERY);
```
Comment (RaheemBux) [inline src/daoimp/UserDAOImpl.java:37]: rename the object name con to connection and ps to preparedStatement

File: src/daoimp/UserDAOImpl.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    @Override
+    public User getUserById(Integer id) {
+        try {
+            Connection conn = DbConnection.getConnection();
+            PreparedStatement ps = conn.prepareStatement(SELECT_QUERY_SINGLE);
+            ps.setInt(1, id);
+            ResultSet rst = ps.executeQuery();
+
+            User u = new User();
+            while (rst.next()) {
```
Comment (RaheemBux) [inline src/daoimp/UserDAOImpl.java:64]: please change object name u to user

File: src/daoimp/UserDAOImpl.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+            e.printStackTrace();
+        }
+        return success;
+    }
+
+    @Override
+    public List<User> getAllUser() {
+
+        List<User> list = new ArrayList<>();
+        try {
```
Comment (RaheemBux) [inline src/daoimp/UserDAOImpl.java:101]: change list to users

File: src/daoimp/UserDAOImpl.java [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+            Connection con = DbConnection.getConnection();
+            PreparedStatement ps = con.prepareStatement(SELECT_QUERY);
+            ResultSet rst = ps.executeQuery();
+
```
Comment (RaheemBux) [inline src/daoimp/UserDAOImpl.java:105]: also change object names here as well

File: src/daoimp/UserDAOImpl.java [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+            while (rst.next()) {
+                User user = new User();
+//                System.out.println(rst.getString("email"));
+                user.setId(rst.getInt("id"));
```
Comment (RaheemBux) [inline src/daoimp/UserDAOImpl.java:109]: remove sop

File: src/daoimp/UserDAOImpl.java [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+            Logger.getLogger(UserDAOImpl.class
+                    .getName()).log(Level.SEVERE, null, ex);
+        }
+        return list;
+    }
+
+    @Override
+    public User getUserIDByName(String name) {
+        throw new UnsupportedOperationException("Not supported yet."); // Generated from nbfs://nbhost/SystemFileSystem/Templates/Classes/Code/GeneratedMethodBody
+    }
```
Comment (RaheemBux) [inline src/daoimp/UserDAOImpl.java:131]: remove the comment from here

File: src/models/User.java [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    private Integer id;
+    private String fName;
+    private String lName;
+    private String phoneNum;
+    private String uEmail;
+    private String password;
+    private Date createdDate;
+    private Date modifiedDate;
+    private boolean status;
+
```
Comment (RaheemBux) [inline src/models/User.java:23]: please create a seperate clas for Auditable fields i.e id, createdDate,modifiedDate, status and also add two more fields createdBy and modifiedBy

File: src/ui/UserFrame.java [hunk 8]
```diff
@@ -0,0 +1,410 @@
...
+        phoneNum.setText("");
+        user_email.setText("");
+        password.setText("");
+    }
+    private User getUserFields(){
+        String fName = fname.getText();
+        String lName = lname.getText();
+        String phNum = phoneNum.getText();
+        String userEmail = user_email.getText();
+        String userPwd = password.getText();
```
Comment (RaheemBux) [inline src/ui/UserFrame.java:71]: please change user_email to userEmail

File: src/ui/UserFrame.java [hunk 9]
```diff
@@ -0,0 +1,410 @@
...
+     * This method is called from within the constructor to initialize the form.
+     * WARNING: Do NOT modify this code. The content of this method is always
+     * regenerated by the Form Editor.
+     */
+    @SuppressWarnings("unchecked")
+    // <editor-fold defaultstate="collapsed" desc="Generated Code">//GEN-BEGIN:initComponents
+    private void initComponents() {
+
+        jLabel1 = new javax.swing.JLabel();
+        backBtn = new javax.swing.JButton();
```
Comment (RaheemBux) [inline src/ui/UserFrame.java:95]: please rename all the labels names

**Pull request merged**